tests/test_ImageCollection.py

```diff
@@ -297,6 +297,23 @@ def test_group_interval_with_interval_and_unit(self, jaxa_rainfall):
         assert grouped.size().getInfo() == 3
         assert ee.ImageCollection(grouped.get(0)).size().getInfo() == 24
 
+    def test_group_interval_drop_empty_collections(self, s2_sr):
+        ic = s2_sr.filterDate("2021-01-01", "2021-01-07")
+        grouped: ee.List = ic.geetools.groupInterval(duration=1, unit="day")
```

Comment:
no need to use type hinting in the variable of the tests, we are less conservative in this area and in any case that will be self detected by mypy (a the method have a return type)

Reply:
Ok, I deleted the type hinting